### PR TITLE
Replace Target with generic request header transformation

### PR DESCRIPTION
**Pull request by josheinhorn**
Target is just a static header to be appended to requests via HTTP headers. While useful, it makes more sense to support this via generic HTTP header modification.

### CODE REVIEW

**Comment by rockspore** [comment]
I think this is a very good idea. It give customers flexibility to configure custom routes (or other filters such as RBAC) in Envoy via this.

Should we give more capability to this transformation such that path rewrites can also be done here?

File: config/config.go
Comment (theganyo) [inline config/config.go:259]: I like this change a lot. A couple questions: 1. Should this be (or also be) in Operation? 2. The next thing would be a RequestBodyTransformation, so I'm wondering if we have a "RequestTransformations" section or if we leave them top-level?

Comment (rockspore) [reply config/config.go:259]: Is ext_authz able to tamper with request body?

Comment (josheinhorn) [reply config/config.go:259]: No ext_authz cannot change the request body, but ext_proc can. I don't have a preferences between having multiple top-level transform sections versus grouping them under a single `RequestTransforms` section. In general I prefer whatever keeps the config the flattest, and I think both are 2 levels deep.

Comment (josheinhorn) [reply config/config.go:259]: And yes, it is already in the Operation (see line 281).

Comment (theganyo) [reply config/config.go:259]: Ok. Grouping or not is fine with since I assume there will only ever be two anyway: Headers and Body.

Comment (josheinhorn) [reply config/config.go:259]: We probably also want to do first-class transforms for special values like URL and HTTP method. That makes me think we _should_ add `HTTPRequestTransforms` as a grouping, with everything within. Lemme update.

Comment (josheinhorn) [reply config/config.go]: PTAL

**Review by rockspore — APPROVED** [review]
I will remove all the json tags or replace them with mapstructure ones in m own PR then.

File: config/config.go
Comment (theganyo) [reply config/config.go]: Looks good. So we're supporting all these transforms for next release?

**Review by theganyo — APPROVED** [review]

**Comment by theganyo** [comment]
> I will remove all the json tags or replace them with mapstructure ones in m own PR then.

@rockspore You could otherwise add that change to this PR.

**Comment by rockspore** [comment]
> > I will remove all the json tags or replace them with mapstructure ones in m own PR then.
> 
> @rockspore You could otherwise add that change to this PR.

I already refactored the new config and new tests into separate files in my ongoing work...I see a chunk of merge conflicts already, so it wouldn't save anything to do it here.

**Comment by theganyo** [comment]
@josheinhorn FYI: We have the PR requestor do the "Squash and merge" and delete the PR branch once approved if they have merge rights. You have the green light. :)

File: config/config.go
Comment (josheinhorn) [reply config/config.go]: If time allows!

**Pull request merged**
